pymatgen/io/lammps/inputs.py

```diff
@@ -33,10 +33,114 @@
 class LammpsInputFile(InputFile):
     """
     Class representing a LAMMPS input settings file, e.g. in.lammps.
+    Allows for LAMMPS input generation in line/stage wise manner. A stage
+    here is defined as a block of LAMMPS input commands usually performing a
+    specific task during the simulation such as energy minimization or
+    NPT/NVT runs. But more broadly, a stage can also be block of LAMMPS
+    input where simulation box is setup, a set of variables are declared or
+    quantities are computed. A comment beginning with '#' is treated as a
+    header for a stage and marks the start of a new stage in the LAMMPS input.
+    Other comments starting with '##' are treated as conventional comments.
     """
 
     def __init__(self, input_settings: OrderedDict = None):
-        self.input_settings = input_settings
+        """
+        Args:
+            input_settings: Ordered Dictionary of LAMMPS input settings.
+        """
+
+        self.nstages = 0
+        self.ncomments = 0
+        self.input_settings = input_settings if input_settings is not None else OrderedDict()
+        self.curr_stage_name = None
+
+    def init_stage(self):
+        """
+        Use this to initiate new stage/black in LAMMPS input file.
+        """
+
+        self.nstages += 1
+
+        stage_name = "stage %s" % self.nstages
+        self._add_stage_name(stage_name)
+
+    def add_comment(self, comment: str, is_stage_header: bool = False) -> None:
+        """
+         Method to add a comment or a stage header.
+
+        Args:
+            comment: Comment string to be added. The comment will be
+                preceded '##' (double hash) in the generated input.
+            is_stage_header: Set this to True, if the comment should be
+                treated as stage/block header to be used at the beginning of
+                each stage/block. Headers are preceded by '#' (single hash)
+                in the generated input.
+        """
+
+        if is_stage_header:
+            self._add_command("header", "# " + comment)
+        else:
+            self.ncomments += 1
+            self._add_command(f"comment_{self.ncomments}", "## " + comment)
+
+    def add_commands(self, commands: Union[str, list, dict], stage_name: str = None):
+        """
+        Adds LAMMPS command/s and its arguments to LAMMPS input file.
+
+        Args:
+            commands: LAMMPS command/s. Can pass single string or list of LAMMPS command
+            with its arguments. Also accepts dictionary of LAMMPS commands are
+            corresponding arguments as key value pairs.
+            stage_name: If a stage name is mentioned, the command is added
+                under that stage block else latest stage is assumed.
```

Comment:
Minor typographical comments: I think it's more conventional to say "command(s)" rather than "command/s" to indicate possible plural. Also the syntax of "Also accepts dictionary of LAMMPS commands are
            corresponding arguments as key value pairs." is not quite right. I think you mean "commands and corresponding arguments as key,value pairs"